nilearn/regions/hierarchical_kmeans_clustering.py

```diff
@@ -15,6 +15,25 @@ def _remove_empty_labels(labels):
     return inverse_vals[labels]
 
 
+def _adjust_small_clusters(list, n_clusters):
+    '''Takes a list of floats summing to n_clusters and try to round it while
+    enforcing rounded list still sum to n_clusters
+    '''
+    list_round = np.round(list).astype(int)
+    if np.sum(list_round) == n_clusters:
+        return list_round
+    elif np.sum(list_round) < n_clusters:
+        while np.sum(list_round) != n_clusters:
+            idx = np.argmax(list - list_round)
+            list_round[idx] += 1
+        return list_round.astype(int)
+    elif np.sum(list_round) > n_clusters:
+        while np.sum(list_round) != n_clusters:
+            idx = np.argmin(list - list_round)
+            list_round[idx] -= 1
+        return list_round.astype(int)
```

Comment:
But you probably need to check that list_round is nonzero ...

Reply:
I agree, and the tests caught that ! Actually, you reviewed an outdated version of the code. The current version also includes `array_round = np.maximum(array_round, np.ones(array.size))`